src/systems/ec/implementations/vanilla/packages/ec-component-form-select/ec-component-form-select.js

```diff
@@ -0,0 +1,289 @@
+    this.textSearch = null;
+    this.textSelectAll = null;
+    this.selectMultiple = null;
+    this.inputContainer = null;
+    this.searchContainer = null;
+
+    // Bind `this` for use in callbacks
+    this.createCheckboxIcon = this.createCheckboxIcon.bind(this);
+    this.createCheckbox = this.createCheckbox.bind(this);
+    this.updateInputValue = this.updateInputValue.bind(this);
+    this.handleClickSelectMultiple = this.handleClickSelectMultiple.bind(this);
+    this.handleClickCheckbox = this.handleClickCheckbox.bind(this);
+    this.handleClickSelectAll = this.handleClickSelectAll.bind(this);
+  }
+
+  /**
+   * @returns {HTMLElement}
+   */
+  createCheckboxIcon() {
+    const tempElement = document.createElement('div');
+    tempElement.innerHTML = iconSvgUiCheck; // avoiding the use of not-so-stable createElementNs
+    const svg = tempElement.childNodes[0];
+    svg.removeAttribute('height');
+    svg.removeAttribute('width');
+    svg.setAttribute('focusable', false);
+    svg.setAttribute('aria-hidden', true);
+    svg.classList.add('ecl-icon', 'ecl-icon--s', 'ecl-checkbox__icon');
+    return svg;
+  }
+
+  /**
+   * @param {String} id
+   * @param {String} text
+   * @param {String} scope
+   * @param {Function} clickHandler
+   * @returns {HTMLElement}
+   */
+  createCheckbox(id, text, scope) {
+    if (!id || !text || !scope) return '';
+    const checkbox = document.createElement('div');
+    checkbox.classList.add('ecl-checkbox');
+    checkbox.setAttribute('data-select-multiple-value', text);
+    const input = document.createElement('input');
+    input.classList.add('ecl-checkbox__input');
+    input.setAttribute('type', 'checkbox');
+    input.setAttribute('id', `${scope}-${id}`);
+    checkbox.appendChild(input);
+    const label = document.createElement('label');
+    label.classList.add('ecl-checkbox__label');
+    label.setAttribute('for', `${scope}-${id}`);
+    const box = document.createElement('span');
+    box.classList.add('ecl-checkbox__box');
+    box.appendChild(this.createCheckboxIcon());
+    label.appendChild(box);
+    label.appendChild(document.createTextNode(text));
+    checkbox.appendChild(label);
+    return checkbox;
+  }
+
+  /**
+   * Initialise component.
+   */
+  init() {
+    const containerClasses = [
+      'ecl-select__container',
+      'ecl-select__container--m',
+    ];
+    this.textDefault = this.element.getAttribute(this.defaultTextAttribute);
+    this.textSearch = this.element.getAttribute(this.searchTextAttribute);
+    this.textSelectAll = this.element.getAttribute(this.selectAllTextAttribute);
+
+    this.select = queryOne(this.selectMultipleSelector);
+
+    if (this.select.nextSibling.classList.contains('ecl-select__icon')) {
+      this.selectIcon = this.select.nextSibling;
+    }
+
+    this.selectMultiple = document.createElement('div');
+    this.selectMultiple.classList.add('ecl-select__multiple');
+    this.selectMultiple.addEventListener(
```

Comment:
currently, this event is triggered when clicking a checkbox, so it is not possible to select mutliple ones without reopening the overlay every time.
I guess that putting the click event on the first ecl-select__container would be better